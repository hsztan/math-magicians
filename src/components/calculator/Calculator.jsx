import React from 'react';
import calculate from './logic/calculate';
import './calculator.scss';

export default class Calculator extends React.PureComponent {
  constructor() {
    super();
    this.state = { calcObj: { total: null, next: null, operation: null } };
    this.handleClick = this.handleClick.bind(this);
    console.log(this.state);
  }

  // eslint-disable-next-line class-methods-use-this
  handleClick(e) {
    const { calcObj } = this.state;
    const newCalcObj = calculate(calcObj, e.target.innerText);
    this.setState({ calcObj: newCalcObj });
  }

  render() {
    const { calcObj } = this.state;
    return (
      <div className="calculator">
        <div className="calculator__display">{calcObj.total}</div>
        <div className="calculator__keypad">
          <div className="calculator__keypad-row">
            <button
              onClick={this.handleClick}
              type="button"
              className="calculator__keypad-button"
            >
              AC
            </button>
            <button type="button" className="calculator__keypad-button">
              +/-
            </button>
            <button type="button" className="calculator__keypad-button">
              %
            </button>
            <button type="button" className="calculator__keypad-button">
              &divide;
            </button>
          </div>
          <div className="calculator__keypad-row">
            <button type="button" className="calculator__keypad-button">
              7
            </button>
            <button type="button" className="calculator__keypad-button">
              8
            </button>
            <button type="button" className="calculator__keypad-button">
              9
            </button>
            <button type="button" className="calculator__keypad-button">
              &times;
            </button>
          </div>
          <div className="calculator__keypad-row">
            <button type="button" className="calculator__keypad-button">
              4
            </button>
            <button type="button" className="calculator__keypad-button">
              5
            </button>
            <button type="button" className="calculator__keypad-button">
              6
            </button>
            <button type="button" className="calculator__keypad-button">
              -
            </button>
          </div>
          <div className="calculator__keypad-row">
            <button type="button" className="calculator__keypad-button">
              1
            </button>
            <button type="button" className="calculator__keypad-button">
              2
            </button>
            <button type="button" className="calculator__keypad-button">
              3
            </button>
            <button type="button" className="calculator__keypad-button">
              +
            </button>
          </div>
          <div className="calculator__keypad-row">
            <button
              id="zero"
              type="button"
              className="calculator__keypad-button"
            >
              0
            </button>
            <button type="button" className="calculator__keypad-button">
              .
            </button>
            <button type="button" className="calculator__keypad-button">
              =
            </button>
          </div>
        </div>
      </div>
    );
  }
}
