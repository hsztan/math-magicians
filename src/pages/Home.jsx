const Home = () => (
  <div className="main-content">
    <h1>Welcome to our page!</h1>
    <p>
      Lorem ipsum dolor sit amet consectetur, adipisicing elit. Odit, facilis
      excepturi. Accusantium nulla cupiditate voluptatem sapiente aliquid illo
      eum repellendus in quia modi laboriosam vel aperiam fugiat, sequi omnis
      hic.
    </p>
    <p>
      Lorem ipsum dolor sit amet consectetur adipisicing elit. Pariatur harum
      itaque ipsam perspiciatis, asperiores, magnam ex, qui temporibus dolorum
      nisi sequi sunt illo. Quidem libero vitae debitis laborum quod! Itaque.
    </p>
  </div>
);
export default Home;
