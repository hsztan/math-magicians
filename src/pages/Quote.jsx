const Quote = () => <h1>This is Quote Page</h1>;

export default Quote;
