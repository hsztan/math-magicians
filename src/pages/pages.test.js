import renderer from 'react-test-renderer';
import Quote from './Quote';
import Home from './Home';
import Calculator from './CalculatorPage';

describe('Pages renders correctly', () => {
  it('Quote page renders correctly', () => {
    const tree = renderer.create(<Quote />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('Calculator page renders correctly', () => {
    const tree = renderer.create(<Calculator />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('Home page renders correctly', () => {
    const tree = renderer.create(<Home />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
