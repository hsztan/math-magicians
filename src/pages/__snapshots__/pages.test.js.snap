// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pages renders correctly Calculator page renders correctly 1`] = `
<div
  className="main-content calculator"
>
  <div>
    <h1>
      Let's do some math!
    </h1>
  </div>
  <div
    className="calculator"
  >
    <div
      className="calculator__display"
    >
      0
    </div>
    <div
      className="calculator__keypad"
    >
      <div
        className="calculator__keypad-row"
      >
        <button
          className="calculator__keypad-button"
          id=""
          onClick={[Function]}
          type="button"
        >
          AC
        </button>
        <button
          className="calculator__keypad-button"
          id=""
          onClick={[Function]}
          type="button"
        >
          +/-
        </button>
        <button
          className="calculator__keypad-button"
          id=""
          onClick={[Function]}
          type="button"
        >
          %
        </button>
        <button
          className="calculator__keypad-button"
          id=""
          onClick={[Function]}
          type="button"
        >
          ÷
        </button>
      </div>
      <div
        className="calculator__keypad-row"
      >
        <button
          className="calculator__keypad-button"
          id=""
          onClick={[Function]}
          type="button"
        >
          7
        </button>
        <button
          className="calculator__keypad-button"
          id=""
          onClick={[Function]}
          type="button"
        >
          8
        </button>
        <button
          className="calculator__keypad-button"
          id=""
          onClick={[Function]}
          type="button"
        >
          9
        </button>
        <button
          className="calculator__keypad-button"
          id=""
          onClick={[Function]}
          type="button"
        >
          x
        </button>
      </div>
      <div
        className="calculator__keypad-row"
      >
        <button
          className="calculator__keypad-button"
          id=""
          onClick={[Function]}
          type="button"
        >
          4
        </button>
        <button
          className="calculator__keypad-button"
          id=""
          onClick={[Function]}
          type="button"
        >
          5
        </button>
        <button
          className="calculator__keypad-button"
          id=""
          onClick={[Function]}
          type="button"
        >
          6
        </button>
        <button
          className="calculator__keypad-button"
          id=""
          onClick={[Function]}
          type="button"
        >
          -
        </button>
      </div>
      <div
        className="calculator__keypad-row"
      >
        <button
          className="calculator__keypad-button"
          id=""
          onClick={[Function]}
          type="button"
        >
          1
        </button>
        <button
          className="calculator__keypad-button"
          id=""
          onClick={[Function]}
          type="button"
        >
          2
        </button>
        <button
          className="calculator__keypad-button"
          id=""
          onClick={[Function]}
          type="button"
        >
          3
        </button>
        <button
          className="calculator__keypad-button"
          id=""
          onClick={[Function]}
          type="button"
        >
          +
        </button>
      </div>
      <div
        className="calculator__keypad-row"
      >
        <button
          className="calculator__keypad-button"
          id="zero"
          onClick={[Function]}
          type="button"
        >
          0
        </button>
        <button
          className="calculator__keypad-button"
          id=""
          onClick={[Function]}
          type="button"
        >
          .
        </button>
        <button
          className="calculator__keypad-button"
          id=""
          onClick={[Function]}
          type="button"
        >
          =
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`Pages renders correctly Home page renders correctly 1`] = `
<div
  className="main-content"
>
  <h1>
    Welcome to our page!
  </h1>
  <p>
    Lorem ipsum dolor sit amet consectetur, adipisicing elit. Odit, facilis excepturi. Accusantium nulla cupiditate voluptatem sapiente aliquid illo eum repellendus in quia modi laboriosam vel aperiam fugiat, sequi omnis hic.
  </p>
  <p>
    Lorem ipsum dolor sit amet consectetur adipisicing elit. Pariatur harum itaque ipsam perspiciatis, asperiores, magnam ex, qui temporibus dolorum nisi sequi sunt illo. Quidem libero vitae debitis laborum quod! Itaque.
  </p>
</div>
`;

exports[`Pages renders correctly Quote page renders correctly 1`] = `
<div
  className="main-content quote"
>
  <h1>
    Mathematics is not about numbers, equations, computations, or algorithms: it is about understanding. —William Paul Thurston
  </h1>
</div>
`;
